### PR TITLE
Add HDM smart proxy

### PR DESCRIPTION
**Pull request by tuxmea**
HDM smart proxy needs a configuration file, which specifies:
- the HDM URL
- optional HDM user and password

User and password can be omitted, if HDM runs without authentication.

### CODE REVIEW

File: manifests/module/hdm.pp
Comment (ekohl) [inline manifests/module/hdm.pp:1]: The `manifests/module` directory is for modules that are built in. You don't need it for a plugin.

File: templates/plugin/hdm.yml.erb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# HDM Smart Proxy
+---
+:enabled: https
+:hdm_url: '<%= scope.lookupvar("foreman_proxy::plugin::hdm::hdm_url") %>'
+<%- if scope.lookupvar("foreman_proxy::plugin::hdm::hdm_user") and scope.lookupvar("foreman_proxy::plugin::hdm::hdm_password") -%>
+:hdm_user: '<%= scope.lookupvar("foreman_proxy::plugin::hdm::hdm_user") %>'
+:hdm_password: '<%= scope.lookupvar("foreman_proxy::plugin::hdm::hdm_password") %>'
+<%- end -%>
+
```
Comment (ekohl) [inline templates/plugin/hdm.yml.erb:9]:
```suggestion
```

File: templates/plugin/hdm.yml.erb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# HDM Smart Proxy
+---
+:enabled: https
```
Comment (ekohl) [inline templates/plugin/hdm.yml.erb:3]:
```suggestion
:enabled: <%= @module_enabled %>
```

**Review by ekohl — CHANGES_REQUESTED** [review]
In https://github.com/theforeman/puppet-foreman_proxy/tree/master/spec/classes we have some tests for various plugins.

File: manifests/module/hdm.pp
Comment (tuxmea) [reply manifests/module/hdm.pp:1]: File deleted

**Review by ekohl — APPROVED** [review]

**Review by ekohl — CHANGES_REQUESTED** [review]
Looks like lint is unhappy with undocumented parameters.

File: manifests/plugin/hdm.pp [hunk 3]
```diff
@@ -0,0 +1,37 @@
+# = Foreman Proxy HDM plugin
+#
+# This class installs the HDM plugin
+#
+# === Parameters:
+#
+# === Advanced parameters:
+#
+# $hdm_url::                  the API endpoint of your HDM installation
```
Comment (ekohl) [inline manifests/plugin/hdm.pp:9]: Since it's a mandatory parameter, I wouldn't call this advanced. Same with user and password.

**Review by ekohl — CHANGES_REQUESTED** [review]

File: manifests/plugin/hdm.pp (hunk 3 above)
Comment (ekohl) [reply manifests/plugin/hdm.pp:9]: Adding to that: the regular parameters group shows up under `foreman-installer --help` while the advanced group only shows up with `--full-help`.

Comment (tuxmea) [reply manifests/plugin/hdm.pp:7]: done

**Pull request merged**